packages/stream-model-instance-handler/src/schema-utils.ts

```diff
@@ -1,31 +1,42 @@
-import ajv, { SchemaObject } from 'ajv/dist/2020.js'
+import Ajv, { SchemaObject } from 'ajv/dist/2020.js'
 import addFormats from 'ajv-formats'
+import lru from 'lru_map'
 
-/**
- * Simple wrapper around AJV library for doing json-schema validation.
- * TODO: Move schema stream loading out of this.
- */
-export class SchemaValidation {
-  private readonly _validator = new ajv({
+const AJV_CACHE_SIZE = 500
+
+function buildAjv(): Ajv {
+  const validator = new Ajv({
     strict: true,
     allErrors: true,
     allowMatchingProperties: false,
     ownProperties: false,
     unevaluated: false,
   })
+  addFormats(validator)
+  return validator
+}
+
+/**
+ * Simple wrapper around AJV library for doing json-schema validation.
+ * TODO: Move schema stream loading out of this.
+ */
+export class SchemaValidation {
+  readonly validators: lru.LRUMap<string, Ajv>
 
   constructor() {
-    addFormats(this._validator)
+    this.validators = new lru.LRUMap(AJV_CACHE_SIZE)
   }
 
-  public validateSchema(content: Record<string, any>, schema: SchemaObject) {
-    const isValid = this._validator.validate(schema, content)
-
-    // Remove schema from the Ajv instance's cache, otherwise the ajv cache grows unbounded
-    this._validator.removeSchema(schema)
+  public validateSchema(content: Record<string, any>, schema: SchemaObject, schemaId: string) {
```

Comment:
> it'd be nice if schemaId was a CommitID instead of a string.

It only introduces overhead for Tiles: https://github.com/ceramicnetwork/js-ceramic/blob/ee5d72100646dde44cf99d6a22a22543383249f8/packages/stream-tile-handler/src/tile-document-handler.ts#L186 So, good idea, but not this time.

Merging the 3 SchemaValidation classes into one: there is a ticket for that.